### PR TITLE
Event cleanup

### PR DESCRIPTION
**Pull request by elcham**
Cleanup event listeners when they get re-added on update.

### CODE REVIEW

File: src/react-dnd-ax/drag-n-drop-container.js [hunk 1]
```diff
@@ -55,6 +55,7 @@ const DragNDropContainer = (WrappedComponent) => {
     }
 
     componentWillUnmount() {
+      window.removeEventListener('dragover', this.setClientY)
```
Comment (artecher) [inline src/react-dnd-ax/drag-n-drop-container.js]: good one

Comment (artecher) [inline src/react-dnd-ax/drag-n-drop-container.js:58]: good one

**Review by artecher — COMMENTED** [review]
@elcham , in general, I like your format change. But I don't fully get which part of your changes improved the performance. If you have a benchmark of the performance improvement, could you share it with me. Or just share me the repo you are working on, I'd like to try the repo w/ and w/out your changes to see performance changes.

**Comment by elcham** [comment]
The performance improvement comes from multiple interactions with the drag-n-drop-items.

Using the current branch, we noticed the steadily increasing time of execution when clicking on the drag elements of multiple items using Chrome's performance tools: 
master-event-listeners

Click Event Listener Time:
1: 30ms
2: 43ms
3: 81ms
4: 108ms
5: 209ms
6: 347ms
7: 662ms
8: 1180ms

However on the PR branch the same click events execution time change to : 
event-cleanup-listeners

Click Event Listener Time:
1: 18ms
2: 25ms
3: 23ms
4: 22ms
5: 21ms
6: 24ms
7: 25ms
8: 20ms

The root cause of this is that when componentDidUpdate gets called, new event listeners are getting added to the dragPointRef for the click event(as well as other events).  However with javascript, elements can have multiple listeners on elements for the same events.  This means that as a user drags and drops multiple items, they will get duplicate event listeners added to the dragPointRef, which all require memory to resolve.  This can eventually cause the browser to crash.

**Comment by artecher** [comment]
I got it now. Good findings!

**Comment by artecher** [comment]
Merged and npm package version bumped to react-dnd-ax@1.0.7

**Comment by artecher** [comment]
@iterativo , this PR may help improve DnD in Attendance as well.

**Pull request merged**